include/ndarray/pybind11.h

```diff
@@ -55,32 +55,36 @@ template <typename T, int N, int C>
 class type_caster< ndarray::Array<T,N,C> > {
 public:
     bool load(handle src, bool) {
-        ndarray::PyPtr tmp(src.ptr(), true);
+        _src = src; // keep alive for stage 2
+        ndarray::PyPtr tmp(_src.ptr(), true);
         if (!ndarray::PyConverter< ndarray::Array<T,N,C> >::fromPythonStage1(tmp)) {
             PyErr_Clear();
             return false;
         }
-        if (!ndarray::PyConverter< ndarray::Array<T,N,C> >::fromPythonStage2(tmp, value)) {
-            PyErr_Clear();
-            return false;
-        }
         return true;
     }
     static handle cast(const ndarray::Array<T,N,C> &src, return_value_policy /* policy */, handle /* parent */) {
         return ndarray::PyConverter< ndarray::Array<T,N,C> >::toPython(src);
     }
+    void set_value() {
+        ndarray::PyPtr tmp(_src.ptr(), true);
+        if (!ndarray::PyConverter< ndarray::Array<T,N,C> >::fromPythonStage2(tmp, value)) {
+            throw error_already_set();
+        }
+    }
 /* This part is normally created by the PYBIND11_TYPE_CASTER macro, which
  * can't be used here due to the partial specialization
  */
 protected:
+    handle _src;
     ndarray::Array<T,N,C> value;
```

Comment:
This is probably from my commit, but our convention say this should be `_value`, not `value`.